course_info/static/course_info/js/controllers/PeopleController.js

```diff
@@ -14,26 +14,64 @@
 
         // set up functions we'll be calling later
         $scope.addUser = function(searchTerm) {
-            if ($scope.searchResults.length > 1) {
-                // TODO - require that a radio button is checked
-                //        (ie. that $scope.selectedResult is valid)
-                // TODO - we've already searched and presented a choice of
-                // role_type_cd, so just POST the user/role.
+            if ($scope.searchResults.length > 1 && $scope.selectedResult.id) {
+                // TODO: assert that radio box is selected
+                $scope.addUserToCourse(searchTerm,
+                                       {user_id: filteredResults[0].univ_id,
+                                        role_id: $scope.selectedRole.roleId});
             }
             else if ($scope.searchResults === 1) {
-                // TODO - shouldn't get here...with a single result, we should
-                // just be POSTing.  maybe disable search button while search
-                // is in flight?
+                // TODO - shouldn't get here, log the error
+                console.log('Add user button pressed while we have a single search result');
             }
             else {
                 $scope.lookup(searchTerm);
             }
         };
-        $scope.closeSuccess = function(index) {
-            $scope.successes.splice(index, 1);
+        $scope.addUserToCourse = function(searchTerm, user){
+            var url = djangoUrl.reverse('icommons_rest_api_proxy',
+                                        ['api/course/v2/course_instances/'
+                                         + $scope.courseInstanceId + '/people/']);
+            $http.post(url, user)
+                .success(function(data, status, headers, config, statusText) {
+                    if (data.detail) {
+                        // TODO - put more details into this warning, like
+                        //        and user details.
```

Comment:
Yup.  Currently, we get a note in there that the API was unable to add to Canvas.  Need to add logic in there to only add the partialFailure if we know the course has a Canvas site.  But that's the general idea.